tests/integration/targets/latest/tasks/admin_permission.yml

```diff
@@ -0,0 +1,108 @@
+---
+##
+##
+### PYNAUTOBOT_OBJECT_PERMISSIONS
+##
+##
+- name: "PERMISSION 1: Necessary info creation"
+  networktocode.nautobot.admin_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test PERMISSION One
+    enabled: true
+    object_types:
+      - "dcim.device"
+    actions:
+      - view
+    state: present
+  register: test_one
+
+- name: "PERMISSION 1: ASSERT - Necessary info creation"
+  assert:
+    that:
+      - test_one is changed
+      - test_one['diff']['before']['state'] == "absent"
+      - test_one['diff']['after']['state'] == "present"
+      - test_one['permission']['name'] == "Test PERMISSION One"
+      - test_one['msg'] == "permission Test PERMISSION One created"
+
+- name: "PERMISSION 2: Create duplicate"
+  networktocode.nautobot.admin_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: Test PERMISSION One
+    enabled: true
+    actions:
+      - view
+    object_types:
+      - "dcim.device"
+    state: present
+  register: test_two
+
+- name: "PERMISSION 2: ASSERT - Create duplicate"
+  assert:
+    that:
+      - not test_two['changed']
+      - test_two['permission']['name'] == "Test PERMISSION One"
+      - test_two['msg'] == "permission Test PERMISSION One already exists"
+
+- name: "PERMISSION 3: ASSERT - Update"
+  networktocode.nautobot.admin_permission:
+    url: "{{ nautobot_url }}"
+    token: "{{ nautobot_token }}"
+    name: "Test PERMISSION One"
+    enabled: true
+    actions:
+      - add
+    object_types:
+      - "dcim.device"
+    state: present
+  register: test_three
```

Comment:
Please add the `users` and `groups` fields to ensure they work as well.
This will probably require:
-  Adding items to the [nautobot_populate.py](https://github.com/nautobot/nautobot-ansible/blob/develop/tests/integration/nautobot-populate.py) file to create the user and group that you will use
- `set_fact` at the top of this file to look up the user and group
  - this will require you to add the endpoints to the [lookup plugin](https://github.com/nautobot/nautobot-ansible/blob/develop/plugins/lookup/lookup.py#L166)